front/src/applications/stdcmV2/components/StdcmVias.tsx

```diff
@@ -0,0 +1,119 @@
+}: {
+  disabled?: boolean;
+  setCurrentSimulationInputs: React.Dispatch<
+    React.SetStateAction<StdcmSimulationResult['input'] | undefined>
+  >;
+}) => {
+  const { t } = useTranslation('stdcm');
+  const dispatch = useAppDispatch();
+  const { getPathSteps } = useOsrdConfSelectors();
+  const { updatePathSteps, updateViaStopTimeV2 } = useOsrdConfActions() as StdcmConfSliceActions;
+  const pathSteps = useSelector(getPathSteps);
+
+  const intermediatePoints = useMemo(() => pathSteps.slice(1, -1), [pathSteps]);
+
+  const updatePathStepsList = (pathStep: PathStep | null, index: number) => {
+    const newPathSteps = replaceElementAtIndex(pathSteps, index, pathStep);
+    dispatch(updatePathSteps(newPathSteps));
+  };
+
+  const updatePathStepStopTime = (stopTime: string, index: number) => {
+    const pathStepToUpdate = pathSteps[index];
+    if (!pathStepToUpdate) return;
+    dispatch(
+      updateViaStopTimeV2({
+        via: pathStepToUpdate,
+        duration: formatDurationAsISO8601(Number(stopTime) * 60),
+      })
+    );
+  };
+
+  const deletePathStep = (index: number) => {
+    dispatch(updatePathSteps(pathSteps.filter((_, i) => i !== index)));
+  };
+
+  useEffect(() => {
+    setCurrentSimulationInputs((prevState) => ({
+      ...prevState,
+      pathSteps,
+    }));
+  }, [pathSteps]);
+
+  return (
+    <div className="stdcm-v2-vias-list">
+      {intermediatePoints.length > 0 &&
+        intermediatePoints.map((pathStep, index) => {
+          const pathStepId = index + 1;
+          return (
+            <StdcmCard
+              key={pathStepId}
+              name={t('trainPath.vias')}
+              title={
+                <div className="stdcm-v2-via-icons">
+                  <span>
+                    <img src={IntermediatePointIcon} alt="intermediate-point" />
+                  </span>
+                  <button type="button" onClick={() => deletePathStep(pathStepId)}>
+                    {t('translation:common.delete')}
+                  </button>
+                </div>
+              } // TODO: Remove icon and clear button -> replace them by the "numbered point" icon
+              hasTip
+              disabled={disabled}
+            >
+              <div className="stdcm-v2-vias">
+                <StdcmOperationalPoint
+                  updatePoint={(e) => updatePathStepsList(e, pathStepId)}
+                  point={pathStep}
+                  disabled={disabled}
+                />
+              </div>
+              {pathStep && (
+                <StdcmInputVia
+                  pathStep={pathStep}
+                  updatePathStepStopTime={(e) => updatePathStepStopTime(e, pathStepId)}
+                />
+              )}
+            </StdcmCard>
+          );
+        })}
+      <StdcmCard hasTip disabled={disabled}>
+        <button
+          type="button"
+          onClick={() => {
+            dispatch(updatePathSteps(addElementAtIndex(pathSteps, pathSteps.length - 1, null)));
```

Comment:
Vias were not supposed to be null at the beginning. Pathfinding "luckily" doesn't break because we added a `compact` in `getPathfindingQuery` function in the pathfinding so typescript doesn't give us a warning. In multiple part of the code (but in operational studies), we assume everything will be ok as long as origin and destination are defined in pathsteps and there isn't any other null values. As long as stdcm and operationalstudies stores are properly separated , everything should be fine but we should be carefull about that.
One fix for that would be to create a state containing the pathSteps where vias can be null but update pathSteps in the store only when the "null via" is updated with an operational point. What do you think ?